routing/cross_mwm_connector_serialization.hpp

```diff
@@ -322,11 +350,74 @@ class CrossMwmConnectorSerializer final
     uint32_t m_version = kLastVersion;
     uint32_t m_numTransitions = 0;
     uint64_t m_sizeTransitions = 0;
+    Weight m_accuracy = kAccuracy;
     serial::CodingParams m_codingParams;
     uint8_t m_bitsPerMask = 0;
     std::vector<Section> m_sections;
   };
 
+  template <typename T, typename Sink>
+  static void WriteDelta(BitWriter<Sink> & writer, T value)
+  {
+    static_assert(std::is_integral<T>::value, "T should be integral type");
+    ASSERT_GREATER(value, 0, ());
+
+    bool const success = coding::DeltaCoder::Encode(writer, static_cast<uint64_t>(value));
+    ASSERT(success, ());
+    UNUSED_VALUE(success);
+  }
+
+  template <typename T, typename Source>
+  static T ReadDelta(BitReader<Source> & reader)
+  {
+    static_assert(std::is_integral<T>::value, "T should be integral type");
+    uint64_t const decoded = coding::DeltaCoder::Decode(reader);
+    if (decoded > numeric_limits<T>::max())
+      MYTHROW(CorruptedDataException,
+              ("Decoded value", decoded, "out of limit", numeric_limits<T>::max()));
+
+    return static_cast<T>(decoded);
+  }
+
+  // Encodes current as delta compared with prev.
+  //
+  // Example:
+  // prev    =       3
+  // current = 1, 2, 3, 4, 5, 6, 7, 8, 9, 10
+  // delta   = 5, 3, 1, 2, 4, 6, 7, 8, 9, 10
+  template <typename T>
+  static T EncodePositivesDelta(T prev, T current)
```

Comment:
Please, don't use the word "NaturalNumber", never, especially if you mean numbers that a strictly greater than zero.